jax/_src/util.py

```diff
@@ -34,16 +34,18 @@
 T = TypeVar("T")
 
 def safe_zip(*args):
-  n = len(args[0])
-  for arg in args[1:]:
-    assert len(arg) == n, 'length mismatch: {}'.format(list(map(len, args)))
+  if len(args) != 0:
+    n = len(args[0])
+    for arg in args[1:]:
+      assert len(arg) == n, 'length mismatch: {}'.format(list(map(len, args)))
   return list(zip(*args))
 
 def safe_map(f, *args):
   args = list(map(list, args))
-  n = len(args[0])
-  for arg in args[1:]:
-    assert len(arg) == n, 'length mismatch: {}'.format(list(map(len, args)))
+  if len(args) != 0:
+    n = len(args[0])
+    for arg in args[1:]:
+      assert len(arg) == n, 'length mismatch: {}'.format(list(map(len, args)))
```

Comment:
Is this change necessary for this PR? It doesn't look like `safe_map` with empty args is used anywhere. Do we need to support that?

Reply:
It's not, happy to revert it, but I thought it made sense to have it behave consistently with `safe_zip` (and the vanilla python `map`) - lmk if you still would like me to revert it.

Reply:
I think @mattjj had opinions on this - I'm going to request a review from him